src/main/java/uk/gov/pay/adminusers/service/UserServices.java

```diff
@@ -126,8 +131,21 @@ public UserServices(UserDao userDao,
         Optional<UserEntity> userEntityOptional = userDao.findByUsername(username);
         return userEntityOptional.map(userEntity -> linksBuilder.decorate(userEntity.toUser()));
     }
-
-
+    
+    public Map<String, List<String>> getAdminUserEmailsForGatewayAccountIds(List<String> gatewayAccountIds) {
+        return gatewayAccountIds.stream().collect(Collectors.toMap(
+                gatewayAccountId -> gatewayAccountId,
+                gatewayAccountId -> {
+                    Optional<List<String>> adminUserEmails = serviceFinder.byGatewayAccountId(gatewayAccountId)
+                            .map(service -> userDao.findByServiceId(service.getId())
```

Comment:
This looks like a reasonable first pass to me, I wonder if given the previous example of 500 we'd want to lean on the database to do this rather than iterating through and piecing it together on the app side, I think it would be n gateway accounts * 2 database queries. This likely wouldn't be an issue for the 500 size but might not be resilient to scaling much beyond that.